### PR TITLE
Additional options docs

### PR DESCRIPTION
**Pull request by benbhall**

### CODE REVIEW

File: options-pattern/docs/netcore21generichost.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+A console service could, for example, get all of its settings from Azure App Configuration (in preview at time of writing) and Azure Key Vault.
+
+```csharp
+private static async Task Main(string[] args)
+    {
+        var builder = new HostBuilder()
+            .UseEnvironment(Environment.GetEnvironmentVariable("ENVIRONMENT"))
+            .ConfigureAppConfiguration((hostingContext, config) =>
+            {
+                var keyVaultAddress = Environment.GetEnvironmentVariable("KEY_VAULT_ADDRESS");
```
Comment (WorkRD) [inline options-pattern/docs/netcore21generichost.md:20]: Might be worth pointing out that this requires the managed service identity set for the app registration and permissions set for the developer running

File: options-pattern/src/DiyConfigExample/StaticConfigRoot.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        private static IConfigurationRoot _instance;
+
+        public static IConfigurationRoot Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new ConfigurationBuilder()
+                        .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
```
Comment (WorkRD) [inline options-pattern/src/DiyConfigExample/StaticConfigRoot.cs:18]: this doesn't build out of the box for me- nuget restore has no issues getting the packages

the proj file has no mention of Microsoft.Extensions.Configuration.Json either.

File: options-pattern/docs/netcore21generichost.md (hunk 1 above)
Comment (benbhall) [reply options-pattern/docs/netcore21generichost.md:20]: Thanks, done.

File: options-pattern/src/DiyConfigExample/StaticConfigRoot.cs (hunk 2 above)
Comment (benbhall) [reply options-pattern/src/DiyConfigExample/StaticConfigRoot.cs:18]: Thanks, resolved.

**Pull request merged**